Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -128,23 +135,39 @@ protected override void OnCreate(Bundle savedInstanceState)
 
         protected override void OnStart()
         {
-            base.OnStart();
             tutorialNeeded = IsTutorialNeeded();
+
+            base.OnStart();
         }
 
-        protected override void OnStop()
+        protected override void OnRestart()
         {
-            // Get rid of TTS.
-            if (tts != null)
+            if (goingFromCartToList)
+            {
+                goingFromCartToList = false;
+                StartListIntent();
+            }
+            else if (goingFromListToCart)
             {
-                //TODO: possibly resume TTS in the OnResume method.
-                tts.Stop();
-                tts.Shutdown();
+                goingFromListToCart = false;
+                StartCartIntent();
             }
-            base.OnStop();
+            else if (!talkBackEnabled)
+            {
+                var message = ConfigurationManager.AppSettings["MainOnRestartMsg"];
+                shopLensTTS.Speak(message, needUserAnswerId);
+            }
+
+            base.OnRestart();
         }
 
-        //TODO: make it so that once we return to the main menu, the voicer greets us and asks questions (OnResume method).
+        protected override void OnStop()
+        {
+            // Stop TTS Speech.
+            shopLensTTS.Stop();
```

Comment:
What if shopLensTTS is null? The Stop method has a ?. operator in it (check the ShopLensTextToSpeech class).